### PR TITLE
Adding xnd packages: libndtypes, ndtypes, libxnd, xnd, libgumath, gumath

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

Packages that have hopes of being a future replacement for numpy.

###### Things done

libgumath: init at unstable-2018-11-27 

libxnd: 0.2.0dev3 -> unstable-2018-11-27 

libndtypes: 0.2.0dev3 -> unstable-2018-11-27 

pythonPackages.ndtypes: init at unstable-2018-11-27 

pythonPackages.xnd: init at unstable-2018-11-27 

pythonPackages.gumath: init at unstable-2018-11-27 

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by costrouc** [comment]
@GrahamcOfBorg build libgumath libndtypes libxnd python3Packages.ndtypes python3Packages.xnd python3Packages.gumath

**Comment by costrouc** [comment]
Looks like aarch64 fails due to roundoff error. Ready for review

**Comment by costrouc** [comment]
I am one of the developers behind xnd and it is a fast moving project right now so I wanted to include the latest "stable" release. But everything works in this relase.

File: pkgs/development/libraries/libgumath/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  configureFlags = [
+    "LD=${stdenv.cc.targetPrefix}cc"
+  ];
+
+  doCheck = true;
+
+  meta = with stdenv.lib; {
+    description = "Library supporting function dispatch on general data containers. C base and Python wrapper";
+    homepage = https://xnd.io/;
+    license = licenses.bsd3;
```
Comment (Mic92) [inline pkgs/development/libraries/libgumath/default.nix]: Do you mind adding yourself as a maintainer here?

Comment (costrouc) [reply pkgs/development/libraries/libgumath/default.nix]: Of course. Can't believe I missed that. Fixed.

File: pkgs/development/python-modules/xnd/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      --replace 'runtime_library_dirs = ["$ORIGIN"]' \
+                'runtime_library_dirs = ["${libndtypes}/lib", "${libxnd}/lib"]' \
+  '';
+
+  postInstall = ''
+    mkdir $out/include
+    cp python/xnd/*.h $out/include
+  '';
+
+  doCheck = true;
```
Comment (FRidh) [inline pkgs/development/python-modules/xnd/default.nix]: is the default

Comment (costrouc) [reply pkgs/development/python-modules/xnd/default.nix]: Fixed.

**Comment by costrouc** [comment]
Bump. Changes made and ready for merge.

**Comment by costrouc** [comment]
@GrahamcOfBorg build python3Packages.gumath

**Comment by costrouc** [comment]
gumath depends on all the new packages thus if it builds all deps build.

File: pkgs/development/libraries/libgumath/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{ stdenv
+, fetchFromGitHub
+, libndtypes
+, libxnd
+}:
+
+stdenv.mkDerivation rec {
+  name = "libgumath";
```
Comment (FRidh) [inline pkgs/development/libraries/libgumath/default.nix]: pname

**Comment by FRidh** [comment]
I really like having this in Nixpkgs, thanks.

**Comment by costrouc** [comment]
@GrahamcOfBorg build python3Packages.gumath

**Comment by costrouc** [comment]
Fixed ready for merge

**Review by teh — COMMENTED** [review]
Would it make sense to add a meta section (with license etc) to the python modules?

**Comment by costrouc** [comment]
Right now they inherit the meta from their c compiled libraries they are based on. Are you suggesting a python decryption for the package?

**Comment by teh** [comment]
Right - I had missed the meta `inherit`. Thanks!

**Pull request merged**